src/components/routerLink.vue

```diff
@@ -15,21 +15,29 @@
     to: string | RegisteredRouteWithParams<T> | ReturnType<RouteMethod>,
   } & RouterPushOptions>()
 
+  defineSlots<{
+    default?: (props: {
+      resolved: string,
+      match: boolean,
+      exactMatch: boolean,
+      isExternal: boolean,
+    }) => unknown,
+  }>()
+
   const router = useRouter()
 
   const route = computed(() => router.find(props.to)?.matched)
 
-  const inMatches = computed(() => !!route.value && router.route.matches.includes(readonly(route.value)))
-  const isMatched = computed(() => !!route.value && router.route.matched === route.value)
+  const match = computed(() => !!route.value && router.route.matches.includes(readonly(route.value)))
+  const exactMatch = computed(() => !!route.value && router.route.matched === route.value)
 
   const classes = computed(() => ({
-    'router-link--match': inMatches.value,
-    'router-link--exact-match': isMatched.value,
+    'router-link--match': match.value,
+    'router-link--exact-match': exactMatch.value,
   }))
 
-  function resolve(): string {
-    return router.resolve(props.to)
-  }
+  const resolved = computed(() => router.resolve(props.to))
+  const isExternal = computed(() => resolved.value.startsWith('http'))
```

Comment:
Idk that we can ensure this is true

Reply:
I thought this might be useful since our router-view supports `to='https://www.google.com'` unlike vue-router. Thinking of how prefect added the little icon, I thought this could be useful. Do you think we should just drop it?

Reply:
I think its useful. I just don't think checking for http is sufficient. Since there's no reason fully qualified url couldn't be passed that's same origin.

Reply:
oh right, but notice this is checking `resolved`, which if given a relative path will return a full qualified URL

[here's a line of a test](https://github.com/kitbagjs/router/blob/component-slot-props/src/components/routerLink.browser.spec.ts#L39) that should prove what I'm saying

Reply:
that doesn't seem right to me 🤔 But even if it is, that means resolved will be fully qualified and won't `resolved.value.startsWith('http')` always be true?

Reply:
oh duh. Fixed